pytensor/tensor/rewriting/linalg.py

```diff
@@ -32,12 +31,12 @@ def transinv_to_invtrans(fgraph, node):
 
 
 @register_stabilize
-@node_rewriter([Dot, Dot22])
```

Comment:
Users may introduce Dot22 directly or call rewrites on a graph that was already partially optimized, so we shouldn't changes this. I don't have an immediate solution for the circular dependency.

Reply:
Does moving the blas rewrites to its own file fix the issue? That would be better I think

Reply:
> Users may introduce Dot22 directly or call rewrites on a graph that was already partially optimized, so we shouldn't change this.

Good point, I'll reintroduce.

> Does moving the blas rewrites to its own file fix the issue? That would be better I think

I think so, but I'm hesitant to introduce such a large change.

A lot simpler, but a little ugly is moving the `tensor.rewriting.linalg` import from `tensor/rewriting/__init__.py` and into `tensor/__init__.py` after `tensor.blas` import. What do you think?

Reply:
I think the blas rewrite file is more logical

Reply:
OK. It's a pretty large change.